src/js/modules/core/services/sweeper/sweeper-service.factory.js

```diff
@@ -0,0 +1,207 @@
+                var approxFee = Math.floor(approxTxSize * feePerKB / 1000);
+
+                var outputValue = totalValue - approxFee;
+
+                if (outputValue > 547) {
+                    rawTransaction.addOutput(options.recipient, outputValue);
+
+                    angular.forEach(inputs, function (input, index) {
+                        var currentPrivKey = resultUTXOs[input.address].priv_key;
+
+                        // Set hashType depending on network
+                        var hashType = bitcoinJS.Transaction.SIGHASH_ALL | (options.network.toLowerCase() == 'bcc' ? bitcoinJS.Transaction.SIGHASH_BITCOINCASHBIP143 : 0);
+
+                        rawTransaction.sign(index, currentPrivKey, null, hashType, input.value);
+                    });
+
+                    return {
+                        rawTx: rawTransaction.build().toHex(),
+                        feePaid: approxFee,
+                        inputCount: inputs.length,
+                        totalValue: totalValue
+                    }
+                } else {
+                    return false;
+                }
+            });
+        }
+
+        function hdDiscover(root, batchSize, accountBatchSize, testnet, accountIdx, UTXOs) {
+            testnet = testnet || 0;
+            accountIdx = accountIdx || 0;
+            UTXOs = UTXOs || {};
+
+            var searchPaths = [];
+            for (var i = accountIdx ; i < accountIdx + accountBatchSize; i++) {
+                searchPaths.push(root.derivePath("m/44\'/" + testnet + "\'/" + i + "\'/0"));
+                searchPaths.push(root.derivePath("m/44\'/" + testnet + "\'/" + i + "\'/1"));
+            }
+
+            return $q.all(searchPaths.map(function (node) {
+                return discover(node, batchSize)
+            }))
+                .then(function (resultLists) {
+
+                    var done = true;
+                    resultLists.map(function (result) {
+                        if(Object.keys(result).length > 0) {
+                            done = false;
+                        }
+                        angular.extend(UTXOs, result);
+                    });
+
+                    if (done) {
+                        return UTXOs;
+                    }
+
+                    return hdDiscover(root, batchSize, accountBatchSize, testnet, accountIdx + accountBatchSize, UTXOs);
+                });
+        }
+
+        function genericHdNodeSweep(root, options) {
+            return launchService.getAccountInfo()
+                .then(function (accountInfo) {
+                    bitcoinDataClient = new BlocktrailBitcoinService({
+                        apiKey: accountInfo.api_key,
+                        apiSecret: accountInfo.api_secret,
+                        network: options.network,
+                        testnet: options.testnet
+                    });
+                    return hdDiscover(root, options.batchSize, options.accountBatchSize, !!options.testnet)
```

Comment:
So getting rid of the the double-not, I'd say, later on `testnet ? 1 : 0` evaluates it anyways